docs/website/docs/dlt-ecosystem/verified-sources/rest_api/basic.md

```diff
@@ -685,22 +702,105 @@ def repositories() -> Generator[Dict[str, Any]]:
+
+```py
+def lower_title(record):
+    record["title"] = record["title"].lower()
+    return record
+
+config: RESTAPIConfig = {
+    "client": {
+        "base_url": "https://api.example.com",
+    },
+    "resources": [
+        {
+            "name": "posts",
+            "endpoint": "posts",
+            "processing_steps": [
+                {"filter": lambda x: x["id"] < 10},
+                {"map": lower_title},
+            ],
+        },
+    ],
+}
+```
+
+In the example above:
+
+- The `filter` step uses a lambda function to include only records where `id` is less than 10.
+- The `map` step applies the `lower_title` function to each remaining record.
+
+#### Using `filter`
+
+The `filter` step allows you to exclude records that do not meet certain criteria. The provided function should return `True` to keep the record or `False` to exclude it:
 
 ```py
 {
-    "name": "issue_comments",
-    "endpoint": {
-        ...
+    "name": "posts",
+    "endpoint": "posts",
+    "processing_steps": [
+        {"filter": lambda x: x["id"] in [10, 20, 30]},
+    ],
+}
+```
+
+In this example, only records with `id` equal to 10, 20, or 30 will be included.
+
+#### Using `map`
+
+The `map` step allows you to modify the records fetched from the API. The provided function should take a record as an argument and return the modified record. For example, to anonymize the `email` field:
+
+```py
+def anonymize_email(record):
+    record["email"] = "REDACTED"
+    return record
+
+config: RESTAPIConfig = {
+    "client": {
+        "base_url": "https://api.example.com",
     },
-    "include_from_parent": ["id", "title", "created_at"],
+    "resources": [
+        {
+            "name": "users",
+            "endpoint": "users",
+            "processing_steps": [
+                {"map": anonymize_email},
+            ],
+        },
+    ],
 }
 ```
 
-This will include the `id`, `title`, and `created_at` fields from the `issues` resource in the `issue_comments` resource data. The name of the included fields will be prefixed with the parent resource name and an underscore (`_`) like so: `_issues_id`, `_issues_title`, `_issues_created_at`.
+#### Combining `filter` and `map`
+
+You can combine multiple processing steps to achieve complex transformations:
+
+```py
+{
+    "name": "posts",
+    "endpoint": "posts",
+    "processing_steps": [
+        {"filter": lambda x: x["id"] < 10},
+        {"map": lower_title},
+        {"filter": lambda x: "important" in x["title"]},
+    ],
+}
+```
+
+:::tip
+#### Best practices
+1. Order matters: Processing steps are applied in the order they are listed. Be mindful of the sequence, especially when combining `map` and `filter`.
+2. Function definition: Define your filter and map functions separately for clarity and reuse.
+3. Use `filter` to exclude records early in the process to reduce the amount of data that needs to be processed.
+4. Combine consecutive `map` steps into a single function for better performance.
```

Comment:
Maybe we can clarify that "performance" here means runtime, not memory or i/o.
```suggestion
4. Combine consecutive `map` steps into a single function for faster execution.
```

Reply:
yes, makes sense!